File: Leetcode Daily Challenge/024-June-2025/015-max-difference-you-can-get-from-changing-an-integer.java
// https://leetcode.com/problems/max-difference-you-can-get-from-changing-an-integer/




class Solution {
    public int maxDiff(int num) {
        String a = Integer.toString(num);
        String b = Integer.toString(num);

        int idx = -1;
        for (int i = 0; i < a.length(); i ++) {
            if (a.charAt(i) != '9') {
                idx = i;
                break;
            }
        }

        if (idx != -1) {
            char ch = a.charAt(idx);
            a = a.replace(ch, '9');
        }

        for (int i = 0; i < b.length(); i ++) {
            if (i > 0 && b.charAt(i) != b.charAt(0) && b.charAt(i) != '0') {
                char ch = b.charAt(i);
                b = b.replace(ch, '0');
                break;
            } else if (i == 0 && b.charAt(i) != '1') {
                char ch = b.charAt(i);
                b = b.replace(ch, '1');
                break;
            }
        }

        return Integer.parseInt(a) - Integer.parseInt(b);
    }
}

File: Leetcode Daily Challenge/024-June-2025/015-max-difference-you-can-get-from-changing-an-integer.cpp
// https://leetcode.com/problems/max-difference-you-can-get-from-changing-an-integer/




#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    // TC : O(log(num))
    // SC : O(log(num))
    int maxDiff(int num) {
        string a = to_string(num);
        string b = to_string(num);

        int idx = a.find_first_not_of('9');
        if (idx != string::npos) {
            char ch = a[idx];
            replace(begin(a), end(a), ch, '9');
        }
        
        for (int i = 0; i < b.length(); i ++) {
            if (i > 0 && b[i] != b[0] && b[i] != '0') {
                char ch = b[i];
                replace(begin(b), end(b), ch, '0');
                break;
            } else if (b[0] != '1'){
                char ch = b[0];
                replace(begin(b), end(b), ch, '1');
                break;
            }
        }

        return stoi(a) - stoi(b);
    }
};